src/components/CurrentUser/CurrentUser.tsx

```diff
@@ -1,12 +1,56 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './CurrentUser.scss';
+import { User } from '../../react-app-env';
+import { getUser } from '../../api/api';
 
-export const CurrentUser: React.FC = () => (
-  <div className="CurrentUser">
-    <h2 className="CurrentUser__title"><span>Selected user: 2</span></h2>
+interface Props {
+  selectedUserId: number,
+  setSelectedUserId: (id: number) => void,
+}
 
-    <h3 className="CurrentUser__name">Ervin Howell</h3>
-    <p className="CurrentUser__email">Shanna@melissa.tv</p>
-    <p className="CurrentUser__phone">010-692-6593 x09125</p>
-  </div>
-);
+export const CurrentUser: React.FC<Props> = ({
+  selectedUserId,
+  setSelectedUserId,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    getUser(selectedUserId)
+      .then(userFromServer => setUser(userFromServer))
+      .catch((error) => {
+        // eslint-disable-next-line no-console
+        console.log(error);
+        setUser(null);
+        setSelectedUserId(0);
+      });
+  }, [selectedUserId]);
+
+  return (
+    user && (
```

Comment:
you have a lot of code after && operator, in this case the best way to do it like this
```suggestion
  }, [selectedUserId]);
  
  if (!user) {
    return 'user not selected'; // or something else here
  }

  return (
```